spec/datetimeformat.html

```diff
@@ -260,19 +301,82 @@
               1. Let _fv_ be FormatNumber(_nf2_, _v_).
               1. If the *length* property of _fv_ is greater than 2, let _fv_ be the substring of _fv_ containing the last two characters.
             1. Else if _f_ is *"narrow"*, *"short"*, or *"long"*, then let _fv_ be a String value representing _f_ in the desired form; the String value depends upon the implementation and the effective locale and calendar of _dateTimeFormat_. If _p_ is *"month"*, then the String value may also depend on whether _dateTimeFormat_ has a [[day]] internal slot. If _p_ is *"timeZoneName"*, then the String value may also depend on the value of the [[inDST]] field of _tm_, and if the implementation does not have a localized representation of _f_, then use _f_ itself.
-            1. Replace the substring of _result_ that consists of *"{"*, p, and *"}"*, with _fv_.
-        1. If _dateTimeFormat_.[[hour12]] is *true*, then
-          1. If _pm_ is *true*, then
-            1. Let _fv_ be an implementation and locale dependent String value representing "post meridiem";
+            1. Add new part record { [[type]]: _p_, [[value]]: _fv_ } as a new element of the list _result_.
+          1. Else if _p_ is equal "ampm", then:
+            1. Let _v_ be the value of _tm_.[[hour]].
+            1. If _v_ is greater than 11, then:
+              1. Let _fv_ be an implementation and locale dependent String value representing "post meridiem";
+            1. Else,
+              1. Let _fv_ be an implementation and locale dependent String value representing "ante meridiem".
+            1. Add new part record { [[type]]: "dayperiod", [[value]]: _fv_ } as a new element of the list _result_.
```

Comment:
Shouldn't this part be named to "dayPeriod" for consistency with other compound names (e.g. "timeZoneName")?